client/js/libs/handlebars/ircmessageparser/parseStyle.js

```diff
@@ -27,27 +38,42 @@ function parseStyle(text) {
 	};
 	resetStyle();
 
+	// When called, this "closes" the current fragment by adding an entry to the
+	// `result` array using the styling information set last time a control code
+	// was met.
 	const emitFragment = () => {
+		// Uses the text fragment starting from the last control code position up to
+		// the current position
 		const textPart = text.slice(start, position);
-		start = position + 1;
 
+		// Filters out all non-style related control codes present in this text
 		const processedText = textPart.replace(controlCodesRx, "");
 
-		if (!processedText.length) {
-			return;
+		if (processedText.length) {
+			// Current fragment starts where the previous one ends, or at 0 if none
+			const fragmentStart = result.length ? result[result.length - 1].end : 0;
```

Comment:
This replaces the `.map` previously done at L124. The "fragment optimizer" was updated to concatenate `end` on top of the text.